nativelib/src/main/scala/scala/scalanative/native/pin.scala

```diff
@@ -0,0 +1,15 @@
+package scala.scalanative
+package native
+
+import scala.annotation.StaticAnnotation
+import scala.annotation.meta.{field, getter, setter}
+
+/** An annotation that is used to indicate that given field
+ *  or method should never be eliminated during link-time
+ *  whole-program dead code elimination as long as enclosing
+ *  class or object is reachable.
+ */
+@field
+@getter
+@setter
+final class pin extends StaticAnnotation
```

Comment:
@sjrd What are your thoughts on this as an API? We need this here to not eliminate fields of `java.lang.String` if it was constructed without calling a constructor (e.g. literals.) But it should be also useful to have sane contract between reflection and link-time dead-code elimination.